ironfish/src/consensus/verifier.ts

```diff
@@ -159,7 +163,7 @@ export class Verifier {
     }
 
     try {
-      return transaction.verify(options)
+      return await transaction.verify(options)
```

Comment:
Adding `await` fixed the issue I encountered with `throw()` not being processed properly.
It seemed (with my limited knowledge of ts) like the correct change to do since the call tree of `verifyTransaction` -> `transaction.verify` -> `workerPool.verify` are all `async` functions and just that `verifyTransaction` lacked `await` when calling one level deeper.

Reply:
Yep, this is a good catch -- Just to reiterate what you were saying, the issue is that without awaiting the promise, the `catch` here won't be hit if the promise happens to throw an error/reject.